### PR TITLE
Solcoder/explain contract

### PR DESCRIPTION
**Pull request by STetsing**
- Added switch for AiI copilot. This enables the inline completion and `///` code generation
- Added tab button for Contract/.sol file summarization & explanation 
- right click on selection -> "Explain this code" using solcoder. get function explanation still works

### CODE REVIEW

**Comment by yann300** [comment]
I feel the loading icon, left to the `AI running...` it too big.
@LianaHus ^

**Comment by yann300** [comment]
it's not really clear what does the `AI copilot` ... and how it differentiate from the " solidity copilot " in the settings

**Comment by yann300** [comment]
> Added switch for AiI copilot. This enables the inline completion and /// code generation

ah so the `solidity copilot` in the settings is meant to be replaced?

File: libs/remix-ui/editor/src/lib/suggestion-service/suggestion-service.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import EventEmitter from 'events'
+
+export class SuggestionService {
```
Comment (yann300) [inline libs/remix-ui/editor/src/lib/suggestion-service/suggestion-service.ts:3]: I see you have another file named `suggestion-service.ts`..
This one or the other should probably be removed.

Comment (STetsing) [reply libs/remix-ui/editor/src/lib/suggestion-service/suggestion-service.ts:3]: > ah so the solidity copilot in the settings is meant to be replaced?
yes 

> I see you have another file named suggestion-service.ts..
This one or the other should probably be removed. 
I think it is till part of the original  browser model

Comment (yann300) [reply libs/remix-ui/editor/src/lib/suggestion-service/suggestion-service.ts:3]: ok, I think it's best to clean up what is no longer used, (copilot in the settings and the duplicate suggestion-service.ts), but perhaps you actually already planned that.

**Comment by yann300** [comment]
remix
getting this when trying the code suggestion feature.

**Comment by yann300** [comment]
when loading Remix, it shows directly `Solidity copilot deactivated!` , is this wanted?

**Comment by yann300** [comment]
When hovering the action which activate the AI features, it says that the model is downloaded. There's no way to know that the model is actually downloaded. I actually get "Unexpected token '<' ,... " (see screenshot above). Does this error means taht the model wasn't downloaded correctly?

**Comment by yann300** [comment]
How should we tell users which features come with enabling the AI? (msg in the tooltip, link to documentation, .. ?)

File: libs/remix-ui/editor/src/lib/remix-ui-editor.tsx [hunk 2]
```diff
@@ -811,8 +834,13 @@ export const EditorUI = (props: EditorUIProps) => {
         currentFunction.current = functionImpl.name
         executeGptGenerateDocumentationAction.label = intl.formatMessage({id: 'editor.generateDocumentation2'}, {name: functionImpl.name})
         gptGenerateDocumentationAction = editor.addAction(executeGptGenerateDocumentationAction)
-        executegptExplainFunctionAction.label = intl.formatMessage({id: 'editor.explainFunction2'}, {name: functionImpl.name})
+        executegptExplainFunctionAction.label = intl.formatMessage({id: 'editor.explainFunction'}, {name: functionImpl.name})
```
Comment (yann300) [inline libs/remix-ui/editor/src/lib/remix-ui-editor.tsx:837]: This update the label to show only "Explain function". Should this keep using `editor.explainFunction2`?

File: libs/remix-ui/renderer/src/lib/renderer.tsx [hunk 3]
```diff
@@ -101,7 +101,20 @@ export const Renderer = ({message, opt = {}, plugin}: RendererProps) => {
             <span className="ml-3 pt-1 py-1" >
               <CopyToClipboard content={messageText} className={` p-0 m-0 far fa-copy ${classList}`} direction={'top'} />
             </span>
-            <span className="border border-success text-success btn-sm" onClick={() => { askGtp() }}>ASK GPT</span>
+            <span
+              className="position-relative text-ai text-sm pl-0 pr-2"
+              style={{fontSize: "x-small", alignSelf: "end"}}
+            >
+              AI
```
Comment (yann300) [inline libs/remix-ui/renderer/src/lib/renderer.tsx:108]: remix
Are you sure about showing this like that. This looks pretty weird @LianaHus

Comment (STetsing) [reply libs/remix-ui/renderer/src/lib/renderer.tsx:108]: @yann300 

> getting this when trying the code suggestion feature

this error originates from the code completion server. This is returned when the server is down 

> when loading Remix, it shows directly Solidity copilot deactivated! , is this wanted?

Yes, this is desired. A log is also prompted whenever the user turns ON the copilot 

> When hovering the action which activate the AI features, it says that the model is downloaded. There's no way to know that the model is actually downloaded. I actually get "Unexpected token '<' ,... " (see screenshot above). Does this error means taht the model wasn't downloaded correctly?

Do you have an image with the "model downloaded"? Anything related to the inbrowser model should be discarded by now


> How should we tell users which features come with enabling the AI? (msg in the tooltip, link to documentation, .. ?) 

Maybe the link to the documentation will be more appropriate. Or get them to know it by discovering it intuitively

**Comment by yann300** [comment]
could you add a matomo call for all the actions? (click on doc, explain code, explain contract, activate AI, deactivate AI, etc...)

**Review by yann300 — APPROVED** [review]

**Pull request merged**